jib-core/src/main/java/com/google/cloud/tools/jib/cache/DefaultCacheStorageWriter.java

```diff
@@ -67,35 +64,18 @@ private WrittenLayer(
    * @throws IOException if an I/O exception occurs
    */
   private static void moveIfDoesNotExist(Path source, Path destination) throws IOException {
+    // If the file already exists, we skip renaming and use the existing file. This happens if a
+    // new layer happens to have the same content as a previously-cached layer.
     if (Files.exists(destination)) {
       return;
     }
 
     try {
-      Files.move(source, destination, StandardCopyOption.ATOMIC_MOVE);
+      Files.move(source, destination);
 
-    } catch (FileAlreadyExistsException ignored) {
-      // If the file already exists, we skip renaming and use the existing file. This happens if a
-      // new layer happens to have the same content as a previously-cached layer.
-      //
-      // Do not attempt to remove the try-catch block with the idea of checking file existence
-      // before moving; there can be concurrent file moves.
-
-    } catch (AtomicMoveNotSupportedException ignored) {
-      try {
-        Files.move(source, destination);
-
-      } catch (FileAlreadyExistsException alsoIgnored) {
-        // Same reasoning
-
-      } catch (DirectoryNotEmptyException ex) {
-        // The file system cannot rename directories, so we must resort to copying the directory.
-        Files.createDirectory(destination);
-        try (Stream<Path> sourceFiles = Files.list(source)) {
-          for (Path sourceFile : sourceFiles.collect(Collectors.toList())) {
-            Files.copy(sourceFile, destination.resolve(sourceFile.getFileName()));
-          }
-        }
+    } catch (FileSystemException ex) {
+      if (!Files.exists(destination)) {
```

Comment:
Perhaps add a TODO to log the destination exists in the future or something about this ignored the `FileSystemException` if the `destination` exists.